src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -213,111 +201,71 @@ namespace ts.codefix {
                         const newName = getNewNameIfConflict(node, collidingSymbolMap);
                         identsToRenameMap.set(symbolIdString, newName.identifier);
                         synthNamesMap.set(symbolIdString, newName);
-                        allVarNames.push({ identifier: newName.identifier, symbol });
-                        addNameToFrequencyMap(collidingSymbolMap, originalName, symbol);
+                        variableNames.push({ identifier: newName.identifier, symbol });
+                        collidingSymbolMap.add(originalName, symbol);
                     }
                     else {
                         const identifier = getSynthesizedDeepClone(node);
                         identsToRenameMap.set(symbolIdString, identifier);
-                        synthNamesMap.set(symbolIdString, createSynthIdentifier(identifier, [], allVarNames.filter(elem => elem.identifier.text === node.text).length/*, numberOfAssignmentsSynthesized: 0*/));

```

Comment:
It should be noted that I _did_ find an additional case where `let` should be used instead of `const`, but the code before this PR [produces the wrong result](https://www.typescriptlang.org/play/#code/GYVwdgxgLglg9mABMAFASkQbwFAEgBOAplCPksMRABYoDktaAdFFYWCikQM4AOCXhAFyIAhmACeGALwA+LHgKFe-QoimJufMAMYArLgnQBuBURJkNSrQJO4AvmhN2gA) (uses `const`):

Reply:
Can you add a test for this if it's not already covered?

Reply:
Sorry, I meant that both master _and_ my PR produce an incorrect result on this pattern. I planned to fix it and add a test for it in a later PR. This PR attempts not to change the existing tested behavior at all (but two bugs were fixed accidentally by cloning identifiers correctly).